modules/mono/glue/GodotSharp/GodotSharp/Core/Array.cs

```diff
@@ -109,38 +150,106 @@ public void Dispose()
+        /// <returns>The new size after adding the object.</returns>
         public int Add(object value) => godot_icall_Array_Add(GetPtr(), value);
 
+        /// <summary>
+        /// Checks if this <see cref="Array"/> contains the given object.
+        /// </summary>
+        /// <param name="value">The item to look for.</param>
+        /// <returns>Whether or not this array contains the given object.</returns>
         public bool Contains(object value) => godot_icall_Array_Contains(GetPtr(), value);
 
+        /// <summary>
+        /// Erases all items from this <see cref="Array"/>.
+        /// </summary>
         public void Clear() => godot_icall_Array_Clear(GetPtr());
 
+        /// <summary>
+        /// Searches this <see cref="Array"/> for an object
+        /// and returns its index or -1 if not found.
+        /// Optionally, the initial search index can be passed.
+        /// This is the same as `find` in GDScript.
+        /// </summary>
+        /// <param name="value">The object to search for</param>
+        /// <returns>The index of the object, or -1 if not found.</returns>
         public int IndexOf(object value) => godot_icall_Array_IndexOf(GetPtr(), value);
 
+        /// <summary>
+        /// Inserts a new object at a given position in the array.
+        /// The position must be a valid position of an existing item,
+        /// or the position at the end of the array.
+        /// Existing items will be moved to the right.
+        /// </summary>
+        /// <param name="index">The index to insert at.</param>
+        /// <param name="value">The object to insert.</param>
         public void Insert(int index, object value) => godot_icall_Array_Insert(GetPtr(), index, value);
 
+        /// <summary>
+        /// Removes the first occurrence of the specified value
+        /// from this <see cref="Array"/>.
+        /// </summary>
+        /// <param name="value">The value to remove.</param>
+        /// <returns>A boolean indicating success or failure.</returns>
         public void Remove(object value) => godot_icall_Array_Remove(GetPtr(), value);
 
+        /// <summary>
+        /// Removes an element from this <see cref="Array"/> by index.
+        /// </summary>
+        /// <param name="index">The index of the element to remove.</param>
         public void RemoveAt(int index) => godot_icall_Array_RemoveAt(GetPtr(), index);
 
         // ICollection
 
+        /// <summary>
+        /// Returns the number of elements in this <see cref="Array"/>.
+        /// This is also known as the size or length of the array.
+        /// </summary>
+        /// <returns>The number of elements.</returns>
         public int Count => godot_icall_Array_Count(GetPtr());
 
+        /// <summary>
+        /// This will always be a reference to the array. It does nothing,
+        /// do not use it. It's needed by <see cref="ICollection"/>.
+        /// </summary>
         public object SyncRoot => this;
```

Comment:
Just like with the `IList` properties, we could hide these `ICollection` properties by implementing them explicitly.